### ISSUE TITLE
Finetuning GLUE from pre-trained model unstability issue

### ISSUE THREAD

**Issue by akanksha2812** [facebookresearch/XLM#70]
Are the given hyper-parameters good enough for fine-tuning GLUE SST-2 task as shown in the Readme example. Multiple trials are giving me drastically different results. For e.g in 1st trial - I see convergence after 10 epochs with valid accuracy 90%. 2nd trial- I see no convergence even after 10 epochs with valid accuracy 50%. 3rd trial - I see it converges but the validation accuracy doesn't cross 80%. I also noticed the similar behavior with IMDB dataset. I tried multiple learning rates but haven't done extensive hyper-parameters search.
The parameter are working fine for XNLI task and I am not seeing this unstability issue.  

Do I need to run it for 250 epochs to see the convergence? Do I need to do the hyper param search for GLUE SST-2 task?

**Comment by aconneau** [comment]
Added more details on fine-tuning here: https://github.com/facebookresearch/XLM#3-fine-tune-a-pretrained-model-on-glue-tasks

Basically sweeping on the batch size and on the learning rates of both the encoder and the projection:
```
    "batch_size": [4, 8]

    "optimizer_e": ["adam,lr=0.00000125", "adam,lr=0.0000025", "adam,lr=0.000005"]

    "optimizer_p": ["adam,lr=0.000005", "adam,lr=0.000025", "adam,lr=0.000125"]
```

Although I did not observe the unstability you mentioned

**Comment by akanksha2812** [comment]
I was using pre-trained cross-lingual language model for fine-tuning. I think that was the issue then. Thanks for sharing

**Comment by aconneau** [comment]
the pretrained XLM model (like MLM-15, or MLM-TLM-15) should still give good results on the GLUE benchmark though (~3 points behind the monolingual XLM_en model). I'm still closing this issue but feel free to re-open if needed.
